tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -1332,17 +1368,19 @@ private[optimizer] abstract class OptimizerCore(
         true
 
       case _ =>
-        arg.tpe.base match {
-          case ClassType("s_Predef$$less$colon$less" | "s_Predef$$eq$colon$eq") =>
-            true
-          case _ =>
-            false
-        }
+        isTypeLikelyOptimizable(arg.tpe)
+    }
+
+    receiverAndArgs.exists(isLikelyOptimizable) || {
+      target.toString == "s_reflect_ClassTag$.apply__jl_Class__s_reflect_ClassTag" &&
+      (receiverAndArgs.tail.head match {
+        case PreTransTree(ClassOf(_), _) => true
+        case _                           => false
+      })
     }
-    receiverAndArgs.exists(isLikelyOptimizable)
   }
 
-  private def inline(allocationSite: Option[AllocationSite],
+  private def inline(allocationSites: Set[AllocationSite],
```

Comment:
Ah but no, my combinatorics are completely wrong (writing it down made me realise).

With a Set, there are (N!) / (N-M)! = N*(N-1)*...*(N-M+1) possibilities. And with a list, there are N^M = N*N*...*N possibilities. Which, since M <<< N, is only marginally smaller. I'll change to a `List`.